Dockerfile Modifications

timway opened the pull request:
- Smaller image size:

```bash
$ podman images
REPOSITORY                           TAG           IMAGE ID      CREATED         SIZE
localhost/openvulnquery              modified      baea2e57ba5d  6 seconds ago   53.4 MB
localhost/openvulnquery              original      44db533e3899  21 minutes ago  96.7 MB
```

- Shorter build times:

```bash
$ time podman build -t openvulnquery:modified -f Dockerfile

real	0m20.183s
user	0m10.956s
sys	0m5.554s

$ time podman build -t openvulnquery:original -f Dockerfile

real	5m42.923s
user	6m57.954s
sys	1m27.090s
```

- Python 3.7.0 => Python 3.7.9 improvements (security and otherwise, [](https://docs.python.org/release/3.7.9/whatsnew/changelog.html#changelog))
- Easier to maintain Dockerfile that doesn't link the code to a particular published release. If it's desirable to get an image for a specific release that commit should be checked out and built (going forward).

santosomar commented:
It looks like the actual installation of the openVulnQuery client was missing from your changes:
```
# Finally installing the openVulnQuery client.
RUN set -ex; \
  python3 -m pip install openVulnQuery
```

timway commented:
Nope, the line where I install the wheels from the build container should take care of it. Installing from pip would get you only the version published there vs the versus of code the Dockerfile lives with.

timway commented:
Bumped the Python version from 3.7.9 to 3.7.10 (security fixes). Also below I'll put a copy of the container build. You'll notice in a portion of it I dropped the forced echo of the CLI and break on error with `&&` instead. If you prefer the other style we can amend it to have that.

```
$ time podman build -t openvulnquery:modified -f Dockerfile
STEP 1: FROM python:3.7.10-alpine AS builder
STEP 2: WORKDIR /build
--> c502bbaf05c
STEP 3: COPY . .
--> f0fcbb80f1f
STEP 4: RUN python3 setup.py bdist_wheel
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/openVulnQuery
copying openVulnQuery/__init__.py -> build/lib/openVulnQuery
copying openVulnQuery/main.py -> build/lib/openVulnQuery
creating build/lib/openVulnQuery/_library
copying openVulnQuery/_library/constants.py -> build/lib/openVulnQuery/_library
copying openVulnQuery/_library/query_client.py -> build/lib/openVulnQuery/_library
copying openVulnQuery/_library/config.py -> build/lib/openVulnQuery/_library
copying openVulnQuery/_library/rest_api.py -> build/lib/openVulnQuery/_library
copying openVulnQuery/_library/__init__.py -> build/lib/openVulnQuery/_library
copying openVulnQuery/_library/main.py -> build/lib/openVulnQuery/_library
copying openVulnQuery/_library/cli_api.py -> build/lib/openVulnQuery/_library
copying openVulnQuery/_library/utils.py -> build/lib/openVulnQuery/_library
copying openVulnQuery/_library/authorization.py -> build/lib/openVulnQuery/_library
copying openVulnQuery/_library/_compatibility.py -> build/lib/openVulnQuery/_library
copying openVulnQuery/_library/advisory.py -> build/lib/openVulnQuery/_library
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/openVulnQuery
creating build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/constants.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/query_client.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/config.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/rest_api.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/__init__.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/main.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/cli_api.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/utils.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/authorization.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/_compatibility.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/_library/advisory.py -> build/bdist.linux-x86_64/wheel/openVulnQuery/_library
copying build/lib/openVulnQuery/__init__.py -> build/bdist.linux-x86_64/wheel/openVulnQuery
copying build/lib/openVulnQuery/main.py -> build/bdist.linux-x86_64/wheel/openVulnQuery
running install_egg_info
running egg_info
writing openVulnQuery.egg-info/PKG-INFO
writing dependency_links to openVulnQuery.egg-info/dependency_links.txt
writing entry points to openVulnQuery.egg-info/entry_points.txt
writing requirements to openVulnQuery.egg-info/requires.txt
writing top-level names to openVulnQuery.egg-info/top_level.txt
adding license file 'LICENSE' (matched pattern 'LICEN[CS]E*')
reading manifest file 'openVulnQuery.egg-info/SOURCES.txt'
writing manifest file 'openVulnQuery.egg-info/SOURCES.txt'
Copying openVulnQuery.egg-info to build/bdist.linux-x86_64/wheel/openVulnQuery-1.31-py3.7.egg-info
running install_scripts
adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
creating build/bdist.linux-x86_64/wheel/openVulnQuery-1.31.dist-info/WHEEL
creating 'dist/openVulnQuery-1.31-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'openVulnQuery/__init__.py'
adding 'openVulnQuery/main.py'
adding 'openVulnQuery/_library/__init__.py'
adding 'openVulnQuery/_library/_compatibility.py'
adding 'openVulnQuery/_library/advisory.py'
adding 'openVulnQuery/_library/authorization.py'
adding 'openVulnQuery/_library/cli_api.py'
adding 'openVulnQuery/_library/config.py'
adding 'openVulnQuery/_library/constants.py'
adding 'openVulnQuery/_library/main.py'
adding 'openVulnQuery/_library/query_client.py'
adding 'openVulnQuery/_library/rest_api.py'
adding 'openVulnQuery/_library/utils.py'
adding 'openVulnQuery-1.31.dist-info/LICENSE'
adding 'openVulnQuery-1.31.dist-info/METADATA'
adding 'openVulnQuery-1.31.dist-info/WHEEL'
adding 'openVulnQuery-1.31.dist-info/entry_points.txt'
adding 'openVulnQuery-1.31.dist-info/top_level.txt'
adding 'openVulnQuery-1.31.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
--> f5cb618a7f1
STEP 5: FROM python:3.7.10-alpine
STEP 6: COPY --from=builder /build/dist/*.whl /whl/
--> 1e723958534
STEP 7: RUN python3 -m pip --no-cache-dir install /whl/*.whl     && rm -rf /whl
Processing /whl/openVulnQuery-1.31-py3-none-any.whl
Collecting requests>=2.10.0
  Downloading requests-2.25.1-py2.py3-none-any.whl (61 kB)
Collecting argparse>=1.4.0
  Downloading argparse-1.4.0-py2.py3-none-any.whl (23 kB)
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.4-py2.py3-none-any.whl (153 kB)
Collecting chardet<5,>=3.0.2
  Downloading chardet-4.0.0-py2.py3-none-any.whl (178 kB)
Collecting certifi>=2017.4.17
  Downloading certifi-2020.12.5-py2.py3-none-any.whl (147 kB)
Collecting idna<3,>=2.5
  Downloading idna-2.10-py2.py3-none-any.whl (58 kB)
Installing collected packages: urllib3, idna, chardet, certifi, requests, argparse, openVulnQuery
Successfully installed argparse-1.4.0 certifi-2020.12.5 chardet-4.0.0 idna-2.10 openVulnQuery-1.31 requests-2.25.1 urllib3-1.26.4
--> 8937bbddb0f
STEP 8: CMD ["/usr/local/bin/python3"]
STEP 9: COMMIT openvulnquery:modified
--> 84590754129
84590754129533a1f38043f3dffb4bdb590e23856a6b31225a82a32f8d18bd54

real	0m17.652s
user	0m9.623s
sys	0m6.330s
```

santosomar commented:
Thank you so much!
